torch/onnx/symbolic_helper.py

```diff
@@ -667,8 +667,15 @@ def _select_helper(g: jit_utils.GraphContext, self, dim, index, apply_reshape=Tr
                 g, index, g.op("Constant", value_t=torch.LongTensor([1]))
             )
 
-    index_scalar_type = index.type().scalarType()
-    if index_scalar_type is None or index_scalar_type not in {"Long", "Int"}:
+    try:
+        index_scalar_type = _type_utils.JitScalarType.from_value(index)
+    except errors.OnnxExporterError:
+        index_scalar_type = None
```

Comment:
Wondering if we can have less try catches in our logic

Reply:
It is either 1) try-except on caller-side or 2) `from_value(...,raises=)` parameter to hide it on the implementation.
The PR originally implemented 2), but you've requested 1), which I did to simplify the API signature.

The reason for this is that the `RuntimeError` only manifests when `torch._C.Value` methods (such as `.type().dtype()` or `.type().scalarName()`) are dereferenced in runtime. In fact, that is why `from_name` and `from_dtype` for non hard-coded values are dangerous.

After `_is_in_type_group` was removed, `from_dtype` can even be changed to private, as we don't need it to use it directly; it is only an implementation detail for `from_value`

Reply:
Thanks. I am thinking if it's possible to not raise anywhere and just check for None after from_value. I will take a closer look later

Reply:
It is not usual for OO languages to return None when object creation fail. Exceptions are pretty much the norm

Reply:
using `from_value(value, default=None)` as suggested